tfhe/src/high_level_api/integers/signed/base.rs

```diff
@@ -150,6 +170,30 @@ where
     {
         self.ciphertext.decrypt_trivial()
     }
+
+    pub fn safe_deserialize_conformant<R>(
+        reader: R,
+        serialized_size_limit: u64,
+        sk: &ServerKey,
+    ) -> Result<Self, String>
+    where
+        R: std::io::Read,
+        Id: serde::de::DeserializeOwned,
+    {
+        let parameter_set = FheIntConformanceParams {
+            params: RadixCiphertextConformanceParams {
+                shortint_params: sk.key.pbs_key().key.conformance_params(),
+                num_blocks_per_integer: Id::num_blocks(sk.key.pbs_key().message_modulus()),
+            },
+            id: PhantomData,
+        };
+
+        crate::safe_deserialization::safe_deserialize_conformant(
+            reader,
+            serialized_size_limit,
+            &parameter_set,
+        )
+    }
```

Comment:
With this changes, a verifier needs a server_key instead of a much smaller conformance parameters, right?
This does not seem great to me

We could have a method on the server_key to extract the conformance parameters instead

What do you think?

Reply:
problem is the same on the client side without the key, so maybe extract the conformance paramters from the client key in that case ?

Reply:
maybe a trait with something like "Into<ConformanceParameters>"

Reply:
Updated so that it keeps an API that can be used without server_key

Reply:
looks good to me